nodes/fabric.js

```diff
@@ -121,32 +147,166 @@ module.exports = function (RED) {
         }
     }
 
+
+    /**
+     * An event subscriber that subscribes to only one event and closes after a 2sec timeout if required
+     * It also can listen on an interval of blocks, useful in cases you do not want to keep listenning
+     * Sends all the events in an array or one by one in the msg.payload depending on the configuration
+     * @param {peer} peer The peer to connect to
+     * @param {channel} channel The channel on which create the event hub
+     * @param {string} chaincodeName The name of the chaincode
+     * @param {Node} node node
+     * @param {object} msg the msg object
+     * @returns {Promise<Buffer>} promise
+     */
+    async function subscribeToEvent(peer, channel, chaincodeName, node, msg) {
```

Comment:
I think you can combine the two functions. They both do the same thing

Reply:
That is what I wanted to do first, but they work in a slightly different context. The original one (yours) starts right when the flow starts (in node). However, the version I made starts only when it receives a message from the precedent node. I've not checked that in details yet. I'll come back to you if I have issues with it.